File: we-care-app-backend/src/main/java/clinicmangement/com/We_Care/DTO/DoctorPreviousVisitsPage.java
package clinicmangement.com.We_Care.DTO;

import lombok.Data;

import java.util.List;

@Data
public class DoctorPreviousVisitsPage {

    private List<DoctorsVisitsDTOProjection> content;
    private int totalPages;
    private Long totalElements;
    private int size;
    private int number;
    private int numberOfElements;
}

File: we-care-app-frontend/src/app/common-components/models/doctor-previous-visits-page.ts
import { DoctorPreviousVisitDtoProjection } from "./doctor-previous-visit-dto-projection";

export interface DoctorPreviousVisitsPage {
    message: string,
    payload: {
        content: DoctorPreviousVisitDtoProjection [],
        totalPages: number;
        totalElements: number;
        size: number;
        number: number;
        numberOfElements: number
}
}